libnetwork/iptables/iptables.go

```diff
@@ -110,30 +113,9 @@ func initFirewalld() {
 	}
 }
 
-func detectIptables() {
-	path, err := exec.LookPath("iptables")
-	if err != nil {
-		return
-	}
-	iptablesPath = path
-	path, err = exec.LookPath("ip6tables")
-	if err != nil {
-		return
-	}
-	ip6tablesPath = path
-	supportsXlock = exec.Command(iptablesPath, "--wait", "-L", "-n").Run() == nil
-	mj, mn, mc, err := GetVersion()
-	if err != nil {
-		logrus.Warnf("Failed to read iptables version: %v", err)
-		return
-	}
-	supportsCOpt = supportsCOption(mj, mn, mc)
-}
-
 func initDependencies() {
-	probe()
-	initFirewalld()
 	detectIptables()
+	initFirewalld()
```

Comment:
Not sure if it makes a difference, but `initFirewalld()` -> `FirewalldInit()` performs some set up (`signalHandler()` go routine, `setupDockerZone()`

Does order matter for that? (if there's no specific reason you changed the order here, perhaps it's safer to keep the existing order?

Reply:
I'll revert this change, although the order should not be a problem since `detectIptables` only checks what's available on the filesystem.